cqlsh_tests/test_cqlsh.py

```diff
@@ -1810,8 +1810,12 @@ def test_client_warnings(self):
         """
         max_partitions_per_batch = 5
         self.cluster.populate(3)
-        self.cluster.set_configuration_options({
-            'unlogged_batch_across_partitions_warn_threshold': str(max_partitions_per_batch)})
+
+        if self.cluster.version() >= LooseVersion('4.0'):  # batch size thresholds moved to guardrails in 4.0
```

Comment:
why is there `LooseVersion` here, but in other similar places we're just using `4.0`?

Reply:
I would much rather if we had a single source of truth like `hasGuardrails(cluster)` instead of using version checks in multiple places.

Reply:
Overall these changes were done over time in different tickets; here they are being ported all at once, so it stands out more.
I will look into aligning them and putting the code in one place.

Reply:
I added `supports_guardrails` to `dtest_setup.py`